src/Schema/OracleSchemaManager.php

```diff
@@ -320,19 +323,226 @@ private function getQuotedIdentifierName($identifier)
         return $identifier;
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    public function listTables()
+    {
+        $currentDatabase = (string) $this->_conn->getDatabase();
+
+        $columnsData     = $this->groupObjectRecordsByTable(
+            $this->getDatabaseColumns($currentDatabase),
+            'TABLE_NAME'
+        );
+        $foreignKeysData = $this->groupObjectRecordsByTable(
+            $this->getDatabaseForeignKeys($currentDatabase),
+            'TABLE_NAME'
+        );
+        $indexesData     = $this->groupObjectRecordsByTable(
+            $this->getDatabaseIndexes($currentDatabase),
+            'TABLE_NAME'
+        );
+
+        $tables = [];
+
+        foreach (array_keys($columnsData) as $tableName) {
+            $columns = $this->_getPortableTableColumnList(
+                $tableName,
+                '',
+                $columnsData[$tableName]
+            );
+
+            $foreignKeys = [];
+            if (! empty($foreignKeysData[$tableName])) {
```

Comment:
I think it could be simplified to:
```php
$foreignKeys = $this->_getPortableTableForeignKeysList($foreignKeysData[$tableName] ?? []);
```
This way, it's more compact and readable, and the function call isn't something that we have to optimize.

Reply:
done